masq_lib/src/percentage.rs

```diff
@@ -0,0 +1,257 @@
+}
+
+impl Percentage {
+    pub fn new(num: u8) -> Self {
+        match num {
+            0..=100 => Self { per_cent: num },
+            x => panic!("Accepts only range from 0 to 100 but {} was supplied", x),
+        }
+    }
+
+    pub fn of<N>(&self, num: N) -> N
+    where
+        N: From<u8> + CheckedMul + CheckedAdd + CheckedDiv + PartialOrd + Integer + Debug + Copy,
+    {
+        let zero = N::from(0);
+        if num == zero || N::from(self.per_cent) == zero {
+            return zero;
+        }
+
+        let a = match N::from(self.per_cent).checked_mul(&num) {
+            Some(num) => num,
+            None => return self.handle_upper_overflow(num),
+        };
+
+        if a < N::from(10) {
+            return N::from(0);
+        }
+
+        let rounding = if Percentage::should_be_rounded_down(a) {
+            N::from(0)
+        } else {
+            N::from(1)
+        };
+
+        let hundred = N::from(100);
+
+        if a < hundred {
+            rounding
+        } else {
+            a.checked_div(&hundred)
+                .expect("div failed")
+                .checked_add(&rounding)
+                .expect("rounding failed")
+        }
+    }
+
+    pub fn add_percent_to<N>(&self, num: N) -> N
+    where
+        N: From<u8> + CheckedMul + CheckedAdd + CheckedDiv + PartialOrd + Integer + Debug + Copy,
+    {
+        self.of(num).checked_add(&num).unwrap_or_else(|| {
+            panic!(
+                "Overflowed during addition of {} per cent, that is {:?}, to {:?} of type {}.",
+                self.per_cent,
+                self.of(num),
+                num,
+                type_name::<N>()
+            )
+        })
+    }
+
+    pub fn subtract_percent_from<N>(&self, num: N) -> N
+    where
+        N: From<u8>
+            + CheckedMul
+            + CheckedAdd
+            + CheckedSub
+            + CheckedDiv
+            + PartialOrd
+            + Integer
+            + Debug
+            + Copy,
+    {
+        num.checked_sub(&self.of(num))
+            .expect("should never happen by its principle")
+    }
+
+    fn should_be_rounded_down<N>(num: N) -> bool
+    where
+        N: From<u8> + PartialEq + PartialOrd + Mul<N, Output = N> + CheckedMul + Integer + Copy,
+    {
+        let ten = N::from(10);
+        let upper_limit = ten * ten;
+        let enough_limit = ten;
+        if num == upper_limit {
+            true
+        } else if num >= enough_limit {
+            let modulo = mod_floor(num, upper_limit);
+            modulo
+                < N::from(5)
+                    .checked_mul(&ten)
+                    .expect("Couldn't create limit to compare with")
+        } else {
+            unreachable!("Check to prevent numbers with fewer than two digits failed")
+        }
+    }
+
+    fn handle_upper_overflow<N>(&self, num: N) -> N
+    where
+        N: From<u8> + Div<Output = N> + Mul<Output = N>,
+    {
+        num / From::from(100) * N::from(self.per_cent)
+    }
```

Comment:
Add a method to get percents bigger than one hundred, perhaps thousands, millions, etc. Internally, it'd simply use `mod 100` to get how many times we should multiply or divide the input before we use the remainder as 
`Percentage::new(remainder)` and the other methods to compute the absolute remainder to be added to the big counts done a step before.